### DISCUSSION TITLE
windows build failed

### DISCUSSION THREAD

**Discussion by up1415** [AcademySoftwareFoundation/OpenRV#362] [Build Issues] [answered]
"E:\OpenRV\_build\main_executable.vcxproj" (default target) (1) ->
       "E:\OpenRV\_build\src\bin\apps\rv\rv.vcxproj" (default target) (3) ->
       "E:\OpenRV\_build\cmake\dependencies\dependencies.vcxproj" (default target) (21) ->
       "E:\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj" (default target)
       (118) ->
       (CustomBuild target) ->
         C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsof
       t.CppCommon.targets(254,5): error MSB8066: Custom build for 'E:\OpenRV\_build\CMakeFiles\ba9
       056a2e12bbb280c19368b23728de1\RV_DEPS_PYSIDE2-build-flag.rule;E:\OpenRV\_build\CMakeFiles\ed
       214e5487734762d00e67784b1bbf86\python310.dll.rule;E:\OpenRV\_build\CMakeFiles\b2670343e3e6d1
       3858c550f7afcec460\RV_DEPS_PYTHON3-stage-target.rule' exited with code -1073741819. [E:\Open
       RV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]

    2 Warning(s)
    1 Error(s)

Got only one build error with VS 2022 but this blocks me a while. Could someone help?

**Comment by up1415** [comment]
similiar issues as https://github.com/AcademySoftwareFoundation/OpenRV/discussions/67 

But the issue is not solved

**Comment by thc10** [comment, marked as answer]
This problem is due to python within msys has conflict with py7zr. You can test this problem by running python and import py7zr and it will occur segment fault crash.
I solve this problem by use official version of python instead of python within py7zr.
After you finished official version of python, change PATH in msys to let it find new version first:

> where python
 c:\msys64\mingw64\bin\python.exe

> export PATH=[PATH to your python]:$PATH   // e.g. export PATH=/c/Python/Python311:$PATH

> where python
 c:\Python\Python311\python.exe
 c:\msys64\mingw64\bin\python.exe

then you can run `rvmk` command to test if your problem is solved. : )

p.s. another way may solve this problem is to modify OpenRV\cmake\dependencies\python3.cmake: 134 (line number for me)
replace default python to official python:

SET(_pyside2_make_command
    C:/Python/Python311/python "${_pyside2_make_command_script}"
)

**Reply by up1415** [reply]
Yes. This works great. pz7zr is not supported within msys python.

**Reply by mingvfx** [reply]
why i already set path to officail version,still have this error

**Reply by krishnaavril** [reply]
The above fix didn't solved the issue, I still getting the same error, any thoughts?

```       "C:\OpenRV\_build\main_executable.vcxproj" (default target) (1) ->
       "C:\OpenRV\_build\src\bin\apps\rv\rv.vcxproj" (default target) (3) ->
       "C:\OpenRV\_build\cmake\dependencies\dependencies.vcxproj" (default target) (15) ->
       "C:\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj" (default target) (164) ->
       (CustomBuild target) ->
         C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): error MSB8066: Custom build for 'C:\OpenRV\_build\CMakeFiles\2
       7ffb7829657fcab58aa9f832fd8f307\RV_DEPS_PYTHON3-requirements-flag.rule;C:\OpenRV\_build\CMakeFiles\27ffb7829657fcab58aa9f832fd8f307\RV_DEPS_PYSIDE2-build-flag.rule;C:\OpenRV\_build\C
       MakeFiles\6ef89c14370471f72f972cb264110d52\python310.dll.rule;C:\OpenRV\_build\CMakeFiles\dc0b82080efe119cccb8b58ef9885864\RV_DEPS_PYTHON3-stage-target.rule;C:\OpenRV\cmake\dependenc
       ies\CMakeLists.txt' exited with code 1. [C:\OpenRV\_build\cmake\dependencies\RV_DEPS_PYTHON3-stage-target.vcxproj]

    4879 Warning(s)
    1 Error(s)
```
